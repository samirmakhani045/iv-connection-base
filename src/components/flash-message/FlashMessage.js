import React from 'react';
import PropTypes from 'prop-types';

import { useTranslation } from '@studio/hooks/useTranslation';

import { Wrapper, StyledButton } from './FlashMessage.styles';

const FlashMessage = (props) => {
  const { variant, label, withAction, button, ...remainingProps } = props;
  const { label: buttonLabel, onClick, ...remainingButtonProps } = button || {};
  const { t } = useTranslation('components.flashMessage');

  return (
    <Wrapper data-testid="wrapper" variant={variant} withAction={withAction} {...remainingProps}>
      <span data-testid="label">{label}</span>
      {withAction && (
        <StyledButton
          data-testid="button"
          onClick={onClick}
          label={buttonLabel || t('button')}
          variant="light"
          small
          {...remainingButtonProps}
        />
      )}
    </Wrapper>
  );
};

FlashMessage.propTypes = {
  /** Which theme to apply to the flash message */
  variant: PropTypes.oneOf(['default', 'positive', 'alert']),
  /** Text that is visible and readable on the button. Mandatory if no icon */
  label: PropTypes.string.isRequired,
  /** Display a cta button */
  withAction: PropTypes.bool,
  /** Props to pass down to button */
  button: PropTypes.shape({
    /** Text that is visible and readable on the button */
    label: PropTypes.string,
    /** Callback to be fired when button is pressed */
    onClick: PropTypes.func,
    /** Label to describe the button. Useful when button has no text */
    'aria-label': PropTypes.string,
    /** Indicates whether or not this button has a popup */
    'aria-haspopup': PropTypes.string,
  }),
};

FlashMessage.defaultProps = {
  variant: 'default',
  withAction: false,
};

export default FlashMessage;
