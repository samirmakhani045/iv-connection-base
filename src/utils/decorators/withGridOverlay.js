import React from 'react';
import styled from 'styled-components/macro';

import * as colors from '@studio/theme/colors';
import { Grid, Box } from '@studio/components/grid';

const Div = styled.div({
  position: 'relative',
});

const StyledGrid = styled(Grid)({
  height: 'calc(100vh - 2rem)',
  left: '50%',
  pointerEvents: 'none',
  position: 'absolute',
  top: '50vh',
  transform: 'translate(-50%, -50vh)',
  width: '100%',
  zIndex: '1000',
});

const StyledBox = styled(Box)({
  backgroundColor: colors.red,
  height: '100%',
  opacity: '0.1',

  '&:nth-child(odd)': {
    opacity: '0.15',
  },
});

const withGridOverlay = (story, props) => {
  if (!props?.args?.gridOverlay) return <>{story()}</>;

  return (
    <Div>
      {story()}
      <StyledGrid>
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
        <StyledBox />
      </StyledGrid>
    </Div>
  );
};

export default withGridOverlay;
