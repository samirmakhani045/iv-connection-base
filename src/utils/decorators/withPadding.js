import React from 'react';
import styled from 'styled-components/macro';

const Wrapper = styled.div({
  padding: '3rem',
});

const withPadding = (story) => <Wrapper>{story()}</Wrapper>;

export default withPadding;
