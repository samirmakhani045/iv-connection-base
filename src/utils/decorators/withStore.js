import React from 'react';
import { Provider } from 'react-redux';

import configureStore from '@studio/store/configureStore';

const withStore = (story) => {
  const store = configureStore();

  return <Provider store={store}>{story()}</Provider>;
};

export default withStore;
