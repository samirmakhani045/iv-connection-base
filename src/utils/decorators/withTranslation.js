import React from 'react';
import { Provider } from 'react-redux';
import { I18nextProvider } from 'react-i18next';

import i18n from '@studio/i18n';
import localTranslations from '@studio/translations';
import configureStore from '@studio/store/configureStore';
import { flatten } from '@utils/flatten';

const withTranslation = (story) => {
  const store = configureStore();
  i18n.addResources('en', 'translations', flatten(localTranslations));

  return (
    <I18nextProvider i18n={i18n}>
      <Provider store={store}>{story()}</Provider>
    </I18nextProvider>
  );
};

export default withTranslation;
