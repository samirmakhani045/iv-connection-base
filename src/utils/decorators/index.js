import withGridOverlay from './withGridOverlay';
import withPadding from './withPadding';
import withStore from './withStore';
import withTranslation from './withTranslation';

export { withGridOverlay, withPadding, withStore, withTranslation };
