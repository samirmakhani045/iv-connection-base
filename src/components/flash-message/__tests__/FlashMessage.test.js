import React from 'react';
import { render, screen } from '@testing-library/react';

import { withTranslation } from '@utils/decorators';

import FlashMessage from '../FlashMessage';

const defaultProps = {
  label: 'Message',
};

const Test = (props) => withTranslation(() => <FlashMessage {...defaultProps} {...props} />);

describe('<FlashMessage />', () => {
  it('should render correctly', () => {
    const { container } = render(<Test />);

    expect(screen.queryByTestId('wrapper')).toBeInTheDocument();
    expect(screen.queryByTestId('label')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a flash message with variant: positive', () => {
    const { container } = render(<Test variant="positive" />);

    expect(screen.queryByTestId('wrapper')).toBeInTheDocument();
    expect(screen.queryByTestId('label')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a flash message with variant: alert', () => {
    const { container } = render(<Test variant="alert" />);

    expect(screen.queryByTestId('wrapper')).toBeInTheDocument();
    expect(screen.queryByTestId('label')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a flash message with a cta', () => {
    const { container } = render(<Test withAction />);

    expect(screen.queryByTestId('wrapper')).toBeInTheDocument();
    expect(screen.queryByTestId('label')).toBeInTheDocument();
    expect(screen.queryByTestId('button')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
