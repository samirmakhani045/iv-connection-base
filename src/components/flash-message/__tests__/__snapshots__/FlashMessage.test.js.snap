// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlashMessage /> should render a flash message with a cta 1`] = `
<div
  class="FlashMessagestyles__Wrapper-sc-7b132e-0 bmyliG"
  data-testid="wrapper"
>
  <span
    data-testid="label"
  >
    Message
  </span>
  <button
    class="Buttonstyles__StyledButton-sc-1uuuyg7-0 KqqqC FlashMessagestyles__StyledButton-sc-7b132e-1 bzDAPJ"
    data-testid="button"
    type="button"
  >
    <span>
      Done
    </span>
  </button>
</div>
`;

exports[`<FlashMessage /> should render a flash message with variant: alert 1`] = `
<div
  class="FlashMessagestyles__Wrapper-sc-7b132e-0 kotSkp"
  data-testid="wrapper"
>
  <span
    data-testid="label"
  >
    Message
  </span>
</div>
`;

exports[`<FlashMessage /> should render a flash message with variant: positive 1`] = `
<div
  class="FlashMessagestyles__Wrapper-sc-7b132e-0 bZLyiS"
  data-testid="wrapper"
>
  <span
    data-testid="label"
  >
    Message
  </span>
</div>
`;

exports[`<FlashMessage /> should render correctly 1`] = `
<div
  class="FlashMessagestyles__Wrapper-sc-7b132e-0 fZtCyI"
  data-testid="wrapper"
>
  <span
    data-testid="label"
  >
    Message
  </span>
</div>
`;
